frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/ticketGroups.component.tsx

```diff
@@ -16,43 +16,76 @@
 import { formatMessage } from '@/v5/services/intl';
-import { IGroupSettingsForm, Viewpoint, ViewpointState } from '@/v5/store/tickets/tickets.types';
+import { GroupOverride, IGroupSettingsForm, Viewpoint, ViewpointState } from '@/v5/store/tickets/tickets.types';
 import { useEffect, useState } from 'react';
-import { dispatch } from '@/v4/modules/store';
+import { useDispatch, useSelector, useStore } from 'react-redux';
 import { ViewpointsActions } from '@/v4/modules/viewpoints';
 import { viewpointV5ToV4 } from '@/v5/helpers/viewpoint.helpers';
-import { cloneDeep } from 'lodash';
-import { useSelector, useStore } from 'react-redux';
+import { cloneDeep, uniqBy } from 'lodash';
 import { VIEWER_PANELS } from '@/v4/constants/viewerGui';
 import { selectLeftPanels } from '@/v4/modules/viewerGui';
 import { selectHiddenGeometryVisible } from '@/v4/modules/tree/tree.selectors';
+import { TreeActions } from '@/v4/modules/tree';
 import { Container, Popper } from './ticketGroups.styles';
 import { GroupsAccordion } from './groupsAccordion/groupsAccordion.component';
 import { TicketGroupsContextComponent } from './ticketGroupsContext.component';
 import { GroupSettingsForm } from './groups/groupActionMenu/groupSettingsForm/groupSettingsForm.component';
 
+const getAllPrefixesCombinations = (overrides: GroupOverride[] = []): string[][] => {
+	const prefixes = overrides.map(({ prefix }) => (prefix)).filter(Boolean);
+	const uniquePrefixes = uniqBy(prefixes, JSON.stringify);
+	const allPrefixesWithDuplicates: string[][] = [];
+
+	uniquePrefixes.forEach((prefix) => {
+		const usedSegments: string[] = [];
+		prefix.forEach((segment) => {
+			allPrefixesWithDuplicates.push(usedSegments.concat(segment));
+			usedSegments.push(segment);
+		});
+	});
+
+	const allPrefixes = uniqBy(allPrefixesWithDuplicates, JSON.stringify);
+	return allPrefixes.sort();
+};
+
 interface TicketGroupsProps {
 	value?: Viewpoint;
 	onChange?: (newvalue) => void;
 	onBlur?: () => void;
 }
 
 enum OverrideType {
-	COLORED,
-	HIDDEN,
+	COLORED = 'colored',
+	HIDDEN = 'hidden',
 }
 
 export const TicketGroups = ({ value, onChange, onBlur }: TicketGroupsProps) => {
-	const [editingOverride, setEditingOverride] = useState<{index:number, type: OverrideType}>({ index: -1, type: OverrideType.COLORED });
+	const dispatch = useDispatch();
+	const [editingOverride, setEditingOverride] = useState<{ index: number, type: OverrideType }>({ index: -1, type: OverrideType.COLORED });
+	const [highglightedOverride, setHighlightedOverride] = useState<{ index: number, type: OverrideType }>({ index: -1, type: null });
+
 	const state: Partial<ViewpointState> = value.state || {};
 	const leftPanels = useSelector(selectLeftPanels);
 	const isSecondaryCard = leftPanels[0] !== VIEWER_PANELS.TICKETS;
 	const store = useStore();
+	const settingsFormGroups = value.state?.[((editingOverride.type === OverrideType.COLORED) ? 'colored' : 'hidden')];
+
+	const handleSetHighlightedIndex = (type) => (index) => {
```

Comment:
Why is this handle function needed?

Reply:
It's just a shortcut to set the highlighted index. Using currying, I can write 
```ts
setHighlightedIndex={handleSetHighlightedIndex(OverrideType.COLORED)}
```
instead of
```ts
setHighlightedIndex={(index) => handleSetHighlightedIndex({ type: OverrideType.COLORED, index })}
```